service/lib/dinstaller/software/manager.rb

```diff
@@ -112,10 +113,14 @@ def propose
         proposal = Yast::Packages.Proposal(force_reset = false, reinit = false, _simple = true)
         logger.info "proposal #{proposal["raw_proposal"]}"
 
-        solve_dependencies
+        deps_result = solve_dependencies
 
-        # do not return proposal hash, so intentional nil here
-        nil
+        proposal_result(proposal, deps_result)
+      end
+
+      def validate
+        msgs = propose
```

Comment:
I am not sure if calling to `#propose` unconditionally is a good idea. If a D-Bus client calls to Manager#Propose and then Manager#Errors in a row, then `#propose` would be called twice without any real need.

Just as idea: maybe the callbacks could indicate to Manager that the proposal has to be recalculated. So, `#validate` would only call to `#propose` if needed.

Reply:
OK, I changed code and now it is called only when needed with probed check. IT also helps with not showing intermediate errors that will alter gone.